packages/tools/pactjs-cli/src/contract-generate/tests/generate.test.ts

```diff
@@ -0,0 +1,156 @@
+jest.mock('../../utils/retrieveContractFromChain');
+
+const mockedStringContractDefinition = jest.fn().mockReturnValue({
+  modulesWithFunctions: {
+    crankk01: {},
+  },
+});
+const mockedFileContractDefinition = jest.fn().mockReturnValue({
+  modulesWithFunctions: {
+    crankk01: {},
+  },
+});
+const mockedGenerateDts = jest.fn().mockReturnValue([]);
+
+jest.mock('@kadena/pactjs-generator', () => ({
+  ...jest.requireActual('@kadena/pactjs-generator'),
+  StringContractDefinition: mockedStringContractDefinition,
+  FileContractDefinition: mockedFileContractDefinition,
+  generateDts: mockedGenerateDts,
+}));
+
+import { retrieveContractFromChain } from '../../utils/retrieveContractFromChain';
+import { generate } from '../generate';
+import { ContractGenerateOptions } from '..';
+
+import { mockContract } from './mockdata/contract';
+
+import { Command, program } from 'commander';
+
+const mockedRetrieveContractFromChain =
+  retrieveContractFromChain as jest.MockedFunction<
+    typeof retrieveContractFromChain
+  >;
+mockedRetrieveContractFromChain.mockResolvedValue(mockContract);
+
+const mockProgramError = jest.fn() as jest.Mock<typeof program.error>;
+
+// silence console.log
+jest.spyOn(console, 'log').mockImplementation(() => {});
+
+const createAndRunProgram = async (
+  type: 'file' | 'chain',
+  options?: Partial<ContractGenerateOptions>,
+): Promise<void> => {
+  const program = new Command('generate');
+  (program.error as unknown) = mockProgramError;
+  const action = generate(program, '0.0.0');
+  let result;
+
+  if (type === 'file') {
```

Comment:
why don't we just return the result without assigning it to a result? With early returns it becomes easier to understand